docker-java-core/src/main/java/com/github/dockerjava/core/DefaultDockerClientConfig.java

```diff
@@ -357,6 +357,14 @@ public final Builder withDockerHost(String dockerHost) {
             return this;
         }
 
+        /**
+         *  Considered default if dockerHost is set to the default value.
+         */
+        public final boolean hasDefaultDockerHost() {
```

Comment:
let's change it to `isDockerHostSetExplicitly()`, to detect user's intention to override it (even if set to the default value)

Reply:
Alright, will invert the logic (changed it to `usesImplicitDockerHost()` before).
As you can see, the checks are all over the place in `DefaultDockerClientConfig` now, I lack the idea of a clean way to capture all places atm.

Reply:
Have you considered removing `docker.host` from `DEFAULT_PROPERTIES`, using `dockerHost != null` to detect explicit value, and then passing `dockerHost != null ? dockerHost : DEFAULT_DOCKER_HOST` to `DefaultDockerClientConfig` from `build()`?

Reply:
would work nicely if `dockerHost` could be `null` in `Builder` :neutral_face:
But I will check further in this in this direction.

Reply:
why can't it be? :)

Reply:
It can now. However, it changes the contract of `Builder.withDockerHost()`, meaning it will only do parsing of host URI on `build()`.

Reply:
I think we can keep the validation, and call it from `withProperties` only if the value is not null

Reply:
Yep, works nicely.